tests/functional/ionIonBeam/ion_ion_beam1d.py

```diff
@@ -102,68 +102,91 @@ def vth(x):
 
 
 
-def getMode(t, m):
-    # return the mode 'm' of the FFT : we the use m=1 for 1 wavelength in the whole simulation domain
-    return np.absolute(np.fft.fft(t)[m])
-
 def yaebx(x, a, b):
     return a*np.exp(np.multiply(b, x))
 
+
 def growth_b_right_hand(run_path):
     file = os.path.join(run_path, "EM_B.h5")
     times = get_times_from_h5(file)
 
     r = Run(run_path)
-    byz = np.array([])
+    first_mode = np.array([])
 
     from scipy.optimize import curve_fit
 
     for time in times:
-        interp_by, x = r.GetB(time, merged=True, interp='nearest')['By']
-        interp_bz, x = r.GetB(time, merged=True, interp='nearest')['Bz']
+        B_hier = r.GetB(time, merged=True, interp="linear")
+
+        by_interpolator, xyz_finest = B_hier["By"]
+        bz_interpolator, xyz_finest = B_hier["Bz"]
 
-        by = interp_by(x[0])
-        bz = interp_bz(x[0])
+        # remove the last point so that "x" is periodic wo. last point = first point
+        x = xyz_finest[0][:-1]
 
-        mm = getMode(by-1j*bz, 1) # last arg is the mode
+        by = by_interpolator(x)
+        bz = by_interpolator(x)
 
-        byz = np.append(byz, mm)
+        # get the mode 1, as it is the most unstable in a box of length 33
+        mode1 = np.absolute(np.fft.fft(by-1j*bz)[1])
+        first_mode = np.append(first_mode, mode1)
 
-    popt, pcov = curve_fit(yaebx, times, byz, p0=[0.1, 0.1]) # last args are initial guess
+    popt, pcov = curve_fit(yaebx, times, first_mode, p0=[0.08, 0.09])
```

Comment:
do we need to comment on those initial guesses which are quite close to the solution no ?